Clarify agentless and agent-based setup options in for Jenkins CI integration (#22355)
DataDog/documentation: content/en/continuous_integration/pipelines/jenkins.md

{{< /site-region >}}

## Compatibility

> This integration supports both [agentless](https://docs.datadoghq.com/continuous_integration/pipelines/jenkins/?tab=linux#install-the-datadog-jenkins-plugin) and [agent-based](https://docs.datadoghq.com/continuous_integration/pipelines/jenkins/?tab=linux#install-the-datadog-agent-optional) installation. Agentless is recommended; install the Datadog agent for infrastructure metric correlation.

Reviewer: ```suggestion
This integration supports both [agentless](/continuous_integration/pipelines/jenkins/?tab=linux#install-the-datadog-jenkins-plugin) and [agent-based](/continuous_integration/pipelines/jenkins/?tab=linux#install-the-datadog-agent-optional) installation. Agentless is recommended; install the Datadog agent for infrastructure metric correlation.
```

TRACE -> http://<HOST>:<TRACE_PORT>/v0.3/traces
{{< /code-block >}}

## Install the Datadog Agent (optional)

> The Datadog Jenkins plugin can either report metrics through the Datadog Agent or directly to Datadog if an API key is provided. If you don't have a Datadog Agent running on the Jenkins controller instance, Datadog recommends installing it first by following the [Agent installation instructions][14].

Reviewer: Clicking on the `agent-based` link at the top of the page will take the reader directly to this section, skipping the plugin installation instructions. Besides, the plugin installation instructions now say "agentless" in the title, so the users might think that the plugin is not needed for agent-based setup.

Perhaps we could make it more explicit here, that the plugin is required regardless of whether the agentless or the agent-based mode is used

Reviewer 2: ```suggestion
The Datadog Jenkins plugin can either report metrics through the Datadog Agent or directly to Datadog if an API key is provided. If you don't have a Datadog Agent running on the Jenkins controller instance, Datadog recommends installing it first by following the [Agent installation instructions][14]. Whether you choose to use Agentless mode or the Agent-based mode, you are required to use the plugin. 
```

If the Jenkins controller and the Datadog Agent have been deployed to a Kubernetes cluster, Datadog recommends using the [Admission Controller][2], which automatically sets the `DD_AGENT_HOST` environment variable in the Jenkins controller pod to communicate with the local Datadog Agent.

> <div class="alert alert-info"><strong>Note</strong>: Unix domain sockets are not yet supported for sending CI Visibility traces.</div>

Reviewer 2: We try to avoid making time-based commitments like "not yet".

```suggestion
<div class="alert alert-info"><strong>Note</strong>: Sending CI Visibility traces through UNIX domain sockets is not supported.</div>
```

> ### Configure the Agent for logs collection

Reviewer 2: ```suggestion
### Configure the Agent for log collection
```

To enable [collecting logs from your jobs](#enable-job-log-collection), configure the Datadog Agent to listen for logs on a TCP port:

{{< tabs >}}
{{% tab "Linux" %}}

> 1. Add `logs_enabled: true` to your Agent configuration file `datadog.yaml`, or set the `DD_LOGS_ENABLED=true` environment variable.

Reviewer 2: ```suggestion
1. Add `logs_enabled: true` to your Agent's `datadog.yaml` configuration file, or set the `DD_LOGS_ENABLED` environment variable to true.
```

> 2. Create a file at `/etc/datadog-agent/conf.d/jenkins.d/conf.yaml` with the contents below. Make sure that `service` matches the CI instance name provided earlier.

Reviewer 2: ```suggestion
2. Create a file at `/etc/datadog-agent/conf.d/jenkins.d/conf.yaml` with the following. Ensure the `service` matches the CI instance name provided earlier.
```

   ```yaml
   logs:
   - type: tcp
     port: 10518
     service: my-jenkins-instance
     source: jenkins
   ```

> 3. [Restart the Agent][1] for the changes to take effect.
> 
> [1]: /agent/configuration/agent-commands/?tab=agentv6v7#restart-the-agent

Reviewer 2: ```suggestion
3. [Restart the Agent][101] for the changes to take effect.

[101]: /agent/configuration/agent-commands/?tab=agentv6v7#restart-the-agent
```

{{% /tab %}}

{{% tab "Windows" %}}

> 1. Add `logs_enabled: true` to your Agent configuration file `datadog.yaml`, or set the `DD_LOGS_ENABLED=true` environment variable.

Reviewer 2: ```suggestion
1. Add `logs_enabled: true` to your Agent's `datadog.yaml`  configuration file, or set the `DD_LOGS_ENABLED` environment variable to true.
```

> 2. Create a file at `%ProgramData%\Datadog\conf.d\jenkins.d\conf.yaml` with the contents below. Make sure that `service` matches the CI instance name provided earlier.

Reviewer 2: ```suggestion
2. Create a file at `%ProgramData%\Datadog\conf.d\jenkins.d\conf.yaml` with the following. Ensure the `service` matches the CI instance name provided earlier.
```

   ```yaml
   logs:
   - type: tcp
     port: 10518
     service: my-jenkins-instance
     source: jenkins
   ```

> 3. [Restart the Agent][1] for the changes to take effect.
> 
> [1]: /agent/configuration/agent-commands/?tab=agentv6v7#restart-the-agent

Reviewer 2: ```suggestion
3. [Restart the Agent][101] for the changes to take effect.

[101]: /agent/configuration/agent-commands/?tab=agentv6v7#restart-the-agent
```

{{% /tab %}}

{{% tab "macOS" %}}

> 1. Add `logs_enabled: true` to your Agent configuration file `datadog.yaml`, or set the `DD_LOGS_ENABLED=true` environment variable.
> 
> 2. Create a file at `~/.datadog-agent/conf.d/jenkins.d/conf.yaml` with the contents below. Make sure that `service` matches the CI instance name provided earlier.

Reviewer 2: ```suggestion
1. Add `logs_enabled: true` to your Agent's `datadog.yaml` configuration file, or set the `DD_LOGS_ENABLED` environment variable to true.

2. Create a file at `~/.datadog-agent/conf.d/jenkins.d/conf.yaml` with the following contents. Ensure the `service` matches the CI instance name provided earlier.
```

   ```yaml
   logs:
   - type: tcp
     port: 10518
     service: my-jenkins-instance
     source: jenkins
   ```

> 3. [Restart the Agent][1] for the changes to take effect.
> 
> [1]: /agent/configuration/agent-commands/?tab=agentv6v7#restart-the-agent

Reviewer 2: ```suggestion
3. [Restart the Agent][101] for the changes to take effect.

[101]: /agent/configuration/agent-commands/?tab=agentv6v7#restart-the-agent
```

## Enable job log collection

> This is an optional step that enables the collection of job logs. Both agentless and agent-based options are supported.

Reviewer 2: ```suggestion
This is an optional step that enables the collection of job logs. Both Agentless and Agent-based options are supported.
```

7. Save your configuration.
{{% /tab %}}

{{< /tabs >}}

### Enable with configuration-as-code

{{< tabs >}}

{{% tab "Agentless (using an API key) - recommended" %}}

If your Jenkins instance uses the Jenkins [`configuration-as-code`][1] plugin:

1. Modify the configuration YAML for the entry `datadogGlobalConfiguration` to include the following:
    ```yaml
    unclassified:
    datadogGlobalConfiguration:

>         # Update the endpoints if reporting to Datadog sites other that `datadoghq.com`

Reviewer: ```suggestion
        # Update the endpoints if reporting to Datadog sites other than `datadoghq.com`
```

    def j = Jenkins.getInstance()
    def d = j.getDescriptor("org.datadog.jenkins.plugins.datadog.DatadogGlobalConfiguration")

>     // Update the endpoints if reporting to Datadog sites other that `datadoghq.com`

Reviewer: ```suggestion
    // Update the endpoints if reporting to Datadog sites other than `datadoghq.com`
```

    ```
{{% /tab %}}

> {{% tab "Report through the Datadog Agent (recommended)" %}}

Reviewer: ```suggestion
{{% tab "Report through the Datadog Agent" %}}
```

    d.save()
    ```
{{% /tab %}}

{{< /tabs >}}

### Use environment variables

{{< tabs >}}

{{% tab "Agentless (using an API key) - recommended" %}}

1. Add the following environment variables on your Jenkins instance machine:
    ```bash

>     # Update the endpoints if reporting to Datadog sites other that `datadoghq.com`

Reviewer: ```suggestion
    # Update the endpoints if reporting to Datadog sites other than `datadoghq.com`
```
